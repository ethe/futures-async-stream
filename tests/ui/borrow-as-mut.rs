// SPDX-License-Identifier: Apache-2.0 OR MIT

#![feature(generators)]

use futures_async_stream::stream;

#[stream(item = ())]
async fn borrow_mut(a: Vec<u8>) {
    a.clear(); //~ ERROR E0596
}

fn main() {}
