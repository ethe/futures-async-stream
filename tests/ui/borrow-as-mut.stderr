error[E0596]: cannot borrow `a` as mutable, as it is not declared as mutable
 --> tests/ui/borrow-as-mut.rs:7:5
  |
6 | async fn borrow_mut(a: Vec<u8>) {
  |                     - help: consider changing this to be mutable: `mut a`
7 |     a.clear(); //~ ERROR E0596
  |     ^^^^^^^^^ cannot borrow as mutable
